language/move-compiler/src/command_line/compiler.rs

```diff
@@ -214,9 +217,11 @@ impl<'a> Compiler<'a> {
             &compiled_module_named_address_mapping,
         )?;
         let mut compilation_env = CompilationEnv::new(flags);
-        let (source_text, pprog_and_comments_res) =
-            parse_program(&mut compilation_env, maps, targets, deps)?;
-        let res: Result<_, Diagnostics> = pprog_and_comments_res.and_then(|(pprog, comments)| {
+        let (source_text, pprog_res) = parse_program(&mut compilation_env, targets, deps, maps)?;
+        let mut comments = CommentMap::new();
+
+        let res: Result<_, Diagnostics> = pprog_res.and_then(|pprog| {
+            find_comments(&pprog, &mut compilation_env, &mut comments)?;
```

Comment:
Maybe this is a `TODO`. It would be greate to move this only to `doc-gen` command.